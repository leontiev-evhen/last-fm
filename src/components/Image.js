import React from 'react';
import PropTypes from 'prop-types';
import imagenotfound from '../assets/imagenotfound.png';

export function Image({ path }) {
	if (path) {
		return <img src={path} alt="" />;
	}
	return <img src={imagenotfound} alt="" />;
}

Image.propTypes = {
	path: PropTypes.string,
};
