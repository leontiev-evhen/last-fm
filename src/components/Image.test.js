import React from 'react';
import { mount } from 'enzyme';
import { Image } from './Image';

describe('Image component', () => {
	const props = {
		path: 'path',
	};
	const component = mount(<Image {...props} />);
	it('render correctly Tag component', () => {
		expect(component).toMatchSnapshot();
	});
	it('component has img tag', () => {
		expect(component.find('img')).toHaveLength(1);
	});
	it('check the type of path', () => {
		expect(typeof component.prop('path')).toBe('string');
	});

	describe('Image component has undefined property', () => {
		const nextProps = {
			path: undefined,
		};
		const defaultPathImg = 'imagenotfound.png';
		const component = mount(<Image {...nextProps} />);

		it('component has default image', () => {
			expect(component.find('img').prop('src')).toEqual(defaultPathImg);
		});
	});
});
