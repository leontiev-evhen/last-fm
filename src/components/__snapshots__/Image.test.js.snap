// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component render correctly Tag component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Image
    path="path"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "path": "path",
    },
    "ref": null,
    "rendered": Object {
      "instance": <img
        alt=""
        src="path"
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "src": "path",
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "path": "path",
      },
      "ref": null,
      "rendered": Object {
        "instance": <img
          alt=""
          src="path"
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "path",
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
